src/Profiles.ts

```diff
@@ -1091,6 +1133,66 @@ export class Profiles {
         return filteredProfile;
     }
 
+    public async getCombinedProfile(serviceProfile: IProfileLoaded, baseProfile: IProfileLoaded) {
+        // TODO: This needs to be improved
+        // The idea is to handle all type of ZE Profiles
+        const commonApi = await ZoweExplorerApiRegister.getInstance().getCommonApi(serviceProfile);
+
+        // This check will handle service profiles that have username and password
+        if (serviceProfile.profile.user && serviceProfile.profile.password) {
+            return serviceProfile;
+        }
+
+        // This check is for optional credentials
+        if (baseProfile && serviceProfile.profile.host && serviceProfile.profile.port &&
+            ((baseProfile.profile.host !== serviceProfile.profile.host) && baseProfile.profile.port !== serviceProfile.profile.port)) {
+                return serviceProfile;
+        }
+
+        let combinedSession;
+        if (!commonApi.getSessionFromCommandArgument) {
+            // This is here for extenders
+            combinedSession = ZoweExplorerApiRegister.getInstance().getCommonApi(serviceProfile).getSession();
```

Comment:
should this line be `combinedSession = commonApi.getSession(serviceProfile);` to make use of the commonApi?

Reply:
Can you rename combinedSession to just session? because there is also combinedProfile and its going to be confusing. 

Based on the code profiles are already combined at this point so it is okay to just call it session.